src/utils.rs

```diff
@@ -6,22 +6,36 @@ use serde_json::json;
 use std::{env, path::PathBuf};
 use url::Url;
 
-pub(crate) fn map_path_to_uri(uri: &str) -> Result<String> {
+pub(crate) fn map_path_to_uri(uri_or_sha_prefix: &str) -> Result<String> {
     let uri_has_schema = Regex::new(r"^\w+://").unwrap();
-    if uri_has_schema.is_match(uri) {
-        return Ok(String::from(uri));
+    if uri_has_schema.is_match(uri_or_sha_prefix) {
+        return Ok(String::from(uri_or_sha_prefix));
     }
-    if PathBuf::from(uri).is_absolute() {
-        Ok(format!("file://{uri}"))
+
+    let path = PathBuf::from(uri_or_sha_prefix);
+    if path.exists() {
+        if path.is_absolute() {
+            Ok(format!("file://{uri_or_sha_prefix}"))
+        } else {
+            Ok(format!(
+                "file://{}/{}",
```

Comment:
I fear this would break on windows, because of the hardcoded `/` separator. It would probably be better to use [`std::path::Path::join`](https://doc.rust-lang.org/std/path/struct.Path.html#method.join) to build the full path